distribution/tools/server-cli/src/test/java/org/elasticsearch/server/cli/APMJvmOptionsTests.java

```diff
@@ -62,6 +70,56 @@ public void testFileDeleteWorks() throws IOException {
         assertFalse(Files.exists(tempFile));
     }
 
+    public void testExtractSettings() throws UserException {
+        Settings settings = Settings.builder()
+            .put("tracing.apm.enabled", true)
+            .put("tracing.apm.agent.server_url", "https://myurl:443")
+            .put("tracing.apm.agent.service_node_name", "instance-0000000001")
+            .put("tracing.apm.agent.global_labels.deployment_id", "123")
+            .put("tracing.apm.agent.global_labels.deployment_name", "APM Tracing")
+            .put("tracing.apm.agent.global_labels.organization_id", "456")
+            .build();
+
+        var extracted = APMJvmOptions.extractApmSettings(settings);
+
+        assertThat(
+            extracted,
+            allOf(
+                hasEntry("server_url", "https://myurl:443"),
+                hasEntry("service_node_name", "instance-0000000001"),
+                hasKey("global_labels"), // test that we have collapsed all global labels into one
+                not(hasKey("global_labels.organization_id")) // tests that we strip out the top level label keys
+            )
+        );
+
+        String[] labels = extracted.get("global_labels").split(",");
+
+        assertThat(
+            Arrays.stream(labels).toList(),
+            containsInAnyOrder("deployment_name=APM Tracing", "organization_id=456", "deployment_id=123")
+        );
```

Comment:
I'd probably have done:

```suggestion
        List<String> labels = Arrays.stream(extracted.get("global_labels").split(",")).toList();

        assertThat(labels, hasSize(3));
        assertThat(
            labels,
            containsInAnyOrder("deployment_name=APM Tracing", "organization_id=456", "deployment_id=123")
        );
```